File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.dart
class Solution {
  int largestInteger(List<int> nums, int k) {
    List<int> cnt = List.filled(51, 0);

    for (int x in nums) {
      cnt[x]++;
    }

    if (k == nums.length) {
      return nums.reduce((a, b) => a > b ? a : b);
    }

    if (k == 1) {
      int ans = -1;

      for (int x in nums) {
        if (cnt[x] == 1 && x > ans) {
          ans = x;
        }
      }

      return ans;
    }

    int a = cnt[nums.first] == 1 ? nums.first : -1;
    int b = cnt[nums.last] == 1 ? nums.last : -1;

    return a > b ? a : b;
  }
}

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.c
#include <stdlib.h>

int largestInteger(int* nums, int numsSize, int k)
{
    int cnt[51] = {0};

    for (int i = 0; i < numsSize; i++)
    {
        cnt[nums[i]]++;
    }

    if (k == numsSize)
    {
        int mx = nums[0];

        for (int i = 1; i < numsSize; i++)
        {
            if (nums[i] > mx)
            {
                mx = nums[i];
            }
        }

        return mx;
    }

    if (k == 1)
    {
        int ans = -1;

        for (int i = 0; i < numsSize; i++)
        {
            if (cnt[nums[i]] == 1 && nums[i] > ans)
            {
                ans = nums[i];
            }
        }

        return ans;
    }

    int a = cnt[nums[0]] == 1 ? nums[0] : -1;
    int b = cnt[nums[numsSize - 1]] == 1 ? nums[numsSize - 1] : -1;

    return a > b ? a : b;
}